File: ClientApp/src/app/core/models/appointment.model.ts
export interface iAppointment {
    id: number;
    patientId: number;
    doctorId: number;
    scheduleId: number;
    consultantType: number;
    modeOfConsultant: number;
    dateOfConsultation: any;
    appointmentDate?: any;
    appointmentStatus: string;
    closedBy?: string;
    closedDate?: any;
    symtoms?: string;
    existingIllness?: string;
    drugAllergies?: string;
    note?: string;
    caseNote?: string;
    diagnosis?: string;
    adviceToPatient?: string;
    labTests?: string;
    doctorPhoneNumber: number;
    doctorName: string;
    speciality: string;
    doctorGender: number;
    doctorEmail: string;
    doctorAvatarUrl: string;
    patientName: string;
    patientBirthDay?: any;
    patientPhoneNumber: number;
    patientGender?: number;
    patientEmail: string;
    patientAddress?: string;
}

File: WebAPI/WebAPI/Models/AppointmentModel.cs
using System.Globalization;

namespace WebAPI.Models
{
    public class AppointmentModel
    {
        public int PatientId { get; set; }
        public int DoctorId { get; set; }
        public int ScheduleId { get; set; }
        public int ConsultantType { get; set; }
        public int ModeOfConsultant { get; set; }
        public DateTime DateOfConsultant { get; set; }
        public DateTime AppointmentDate {  get; set; }
        public string? AppointmentStatus { get; set; }
        public string? ClosedBy { get; set; }
        public DateTime ClosedDate { get; set; }
        public string? Symtoms { get; set; }
        public string? Existingillness { get; set; }
        public string? Druggallergies { get; set; }
        public string? Note {  get; set; }
        public string? CaseNote { get; set; }
        public string? Diagnosis { get; set; }
        public string? AdviceToPatient { get; set; }
        public string? LabTests { get; set; }
    }
}
